governance/program/tests/process_cast_vote.rs

```diff
@@ -520,6 +520,286 @@ async fn test_cast_vote_with_vote_tipped_to_defeated() {
+            &proposal_cookie,
+            &token_owner_record_cookie2,
+            YesNoVote::No,
+        )
+        .await
+        .unwrap();
+    let proposal_account = governance_test
+        .get_proposal_account(&proposal_cookie.address)
+        .await;
+    assert_eq!(ProposalState::Voting, proposal_account.state);
+
+    governance_test
+        .with_cast_vote(
+            &proposal_cookie,
+            &token_owner_record_cookie3,
+            YesNoVote::No,
+        )
+        .await
+        .unwrap();
+    let proposal_account = governance_test
+        .get_proposal_account(&proposal_cookie.address)
+        .await;
+    assert_eq!(ProposalState::Voting, proposal_account.state);
+
+    governance_test
+        .with_cast_vote(
+            &proposal_cookie,
+            &token_owner_record_cookie4,
+            YesNoVote::Yes,
+        )
+        .await
+        .unwrap();
+    let proposal_account = governance_test
+        .get_proposal_account(&proposal_cookie.address)
+        .await;
+    assert_eq!(ProposalState::Voting, proposal_account.state);
+
+    // Act: 300 vs 200 makes it tip
+    governance_test
+        .with_cast_vote(
+            &proposal_cookie,
+            &token_owner_record_cookie5,
+            YesNoVote::Yes,
+        )
+        .await
+        .unwrap();
+    let proposal_account = governance_test
+        .get_proposal_account(&proposal_cookie.address)
+        .await;
+    assert_eq!(ProposalState::Succeeded, proposal_account.state);
+    let proposal_owner_record = governance_test
+        .get_token_owner_record_account(&proposal_cookie.account.token_owner_record)
+        .await;
+    assert_eq!(0, proposal_owner_record.outstanding_proposal_count);
+}
+
+#[tokio::test]
+async fn test_cast_vote_with_early_vote_tipped_to_defeated() {
+    // Arrange
+    let mut governance_test = GovernanceProgramTest::start_new().await;
+
+    let realm_cookie = governance_test.with_realm().await;
+    let governed_account_cookie = governance_test.with_governed_account().await;
+
+    let mut governance_config = governance_test.get_default_governance_config();
+
+    governance_config.vote_tipping = VoteTipping::Early;
+    governance_config.vote_threshold_percentage = VoteThresholdPercentage::YesVote(40);
+
+    // 100 votes
+    let token_owner_record_cookie1 = governance_test
+        .with_community_token_deposit(&realm_cookie)
+        .await
+        .unwrap();
+
+    let mut account_governance_cookie = governance_test
+        .with_account_governance_using_config(
+            &realm_cookie,
+            &governed_account_cookie,
+            &token_owner_record_cookie1,
+            &governance_config,
+        )
+        .await
+        .unwrap();
+
+    // 100 votes
+    let token_owner_record_cookie2 = governance_test
+        .with_community_token_deposit(&realm_cookie)
+        .await
+        .unwrap();
+
+    // 100 votes
+    let token_owner_record_cookie3 = governance_test
```

Comment:
nit: Can we capture this use case with just one voter voting No? I think we don't need 3 and can make the test simpler

Reply:
I like that this test confirms that 200 no votes do not tip the vote to defeated, even though it's above 50% and 40% of votes.

Reply:
ok